src/cert_store.rs

```diff
@@ -447,8 +498,89 @@ mod test {
             .unwrap();
     }
 
+    fn find_existing_cert_and_key_helper(pfx: &[u8], pass: &str) { 
+        for _ in 0..2 {
+            let mut identity = None;
+
+            let store = PfxImportOptions::new()
+                .include_extended_properties(true)
+                .password(pass)
+                .import(pfx)
+                .unwrap();
+
+            for cert in store.certs() {
+                if cert
+                    .private_key()
+                    .silent(true)
+                    .compare_key(true)
+                    .acquire()
+                    .is_ok()
+                {
+                    identity = Some(cert);
+                    break;
+                }
+            }
+
+            let mut identity = match identity {
+                Some(identity) => identity,
+                None => {
+                    panic!()
+                }
+            };
+
+            let mut store = CertStore::open_current_user("TestRustMy").unwrap();
+            if let Err(_err) = CertStore::find_existing_cert(&mut store, &identity) {
+                if let Err(_err) = store.add_cert(&identity, CertAdd::Always){
```

Comment:
store.add_cert(&identity, CertAdd::Always).expect("add_cert failed");

.expect() instead of panic!()